webpack/assets/javascripts/react_app/redux/API/APIMiddleware.js

```diff
@@ -8,4 +8,5 @@ export const APIMiddleware = store => next => action => {
   } else {
     next(action);
   }
+  return action;
```

Comment:
I believe we should return return the results of eighter branch, not the action itself.

Reply:
@ezr-ondrej What do you mean by results of either branch?

If we return `next(action)` it should have the same effect, since the return value of `next(action)` will probably be the dispatched action.  If this is what you mean it sounds fine to me.